### PR TITLE
inventory aws_ec2 - add ssm_inventory inventory into hostvars

### LINKED ISSUE #704: [FR] aws_ec2 (dynamic inventory) use "aws ssm get-inventory" info also
### Summary

Hi,

I think `aws_ec2` plugin use ec2 metadata only,
but user who already set up ssm inventroy can use/get ssm inventroy info, too.

When I was searching aws_ec2 plugin,
there's no way to get OS (distro) info,
and `ansible_distribution` can not use.

so I request the way to get ssm inventory or use directly in aws_ec2 plugin.

Thanks,

### Issue Type

Feature Idea

### Component Name

aws_ec2

### Additional Information

```yaml (paste below)

```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by abikouo**
##### SUMMARY

using `use_ssm_inventory` users can populate hostvars with Amazon SSM inventory for configured instances
closes #704 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

`aws_ec2 inventory`

### CODE REVIEW

**Comment by gravesm** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Review by alinabuzachis — COMMENTED** [review]
@abikouo Can you also please update this guide https://github.com/ansible-collections/amazon.aws/blob/main/docs/docsite/rst/aws_ec2_guide.rst with this option?

**Comment by abikouo** [comment]
> @abikouo Can you also please update this guide https://github.com/ansible-collections/amazon.aws/blob/main/docs/docsite/rst/aws_ec2_guide.rst with this option?

done!

File: plugins/inventory/aws_ec2.py [hunk 1]
```diff
@@ -133,6 +133,12 @@
       - The suffix for host variables names coming from AWS.
     type: str
     version_added: 3.1.0
+  use_ssm_inventory:
+    description:
+      - Add SSM inventory information into hostvars.
+    type: bool
+    default: False
+    version_added: 5.3.0
```
Comment (tremble) [inline plugins/inventory/aws_ec2.py:141]:
```suggestion
    version_added: 6.0.0
```

This won't cleanly backport, there's some substantial changes between 5.x and 6.x that I wasn't comfortable backporting.

File: tests/integration/targets/inventory_aws_ec2/playbooks/library/test_get_ssm_inventory.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/python
+# Copyright (c) 2023 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = """
+module: test_get_ssm_inventory
```
Comment (tremble) [inline tests/integration/targets/inventory_aws_ec2/playbooks/library/test_get_ssm_inventory.py:6]: This feels like something we should cleanup and merge into community.aws as `ssm_inventory_info`...

**Review by tremble — CHANGES_REQUESTED** [review]

**Review by tremble — APPROVED** [review]

**Comment by tremble** [comment]
Backporting to 5 will be complicated, I recommend not doing so, given we're less than 2 months from the planned 6.0.0 release.

**Pull request merged**